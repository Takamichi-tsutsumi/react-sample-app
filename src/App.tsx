import * as React from 'react'

export default class App extends React.Component {
  public clickButton = () => {
    console.info('Hello, my name is App')
  }

  public render() {
    return (
      <div>
        <h1>React SSR Sample</h1>
        <p>Hi, How are you??</p>
        <button onClick={this.clickButton}>Hi!</button>
      </div>
    )
  }
}
