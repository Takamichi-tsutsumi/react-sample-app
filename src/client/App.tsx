import * as React from 'react'
import { connect } from 'react-redux'

interface IProps {
  count: number
  increment: () => { type: string }
  decrement: () => { type: string }
}

class App extends React.Component<IProps> {
  public render() {
    return (
      <div>
        <h1>COUNTER</h1>
        <p>{this.props.count}</p>
        <button onClick={this.props.increment}>INCREMENT</button>
        <button onClick={this.props.decrement}>DECREMENT</button>
      </div>
    )
  }
}

export default connect(
  (state: any) => ({
    count: state
  }),
  {
    increment: () => ({ type: 'INCREMENT' }),
    decrement: () => ({ type: 'DECREMENT' })
  }
)(App)
