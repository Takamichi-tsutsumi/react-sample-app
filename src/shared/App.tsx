import * as React from 'react'
import { renderRoutes } from 'react-router-config'
import routes from './routes'
import Layout from './components/Layout'

class App extends React.Component {
  public render() {
    return <Layout>{renderRoutes(routes)}</Layout>
  }
}

export default App
