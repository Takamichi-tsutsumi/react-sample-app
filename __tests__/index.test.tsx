import * as React from 'react'
import { create } from 'react-test-renderer'
import App from '../src/client/App'

test('SnapShot', () => {
  const tree = create(<App />).toJSON()
  expect(tree).toMatchSnapshot()
})
