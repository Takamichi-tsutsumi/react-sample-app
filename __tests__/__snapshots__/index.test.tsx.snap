// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnapShot 1`] = `
<div>
  <h1>
    React SSR Sample
  </h1>
  <p>
    HMR is working fine!!
  </p>
  <button
    onClick={[Function]}
  >
    Hi!
  </button>
</div>
`;
